browser/brave_wallet/brave_wallet_provider_delegate_impl_helper_android.cc

```diff
@@ -4,17 +4,26 @@
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "brave/browser/brave_wallet/brave_wallet_provider_delegate_impl_helper.h"
-
-#include "base/notreached.h"
+#include "brave/browser/brave_wallet/brave_wallet_service_factory.h"
+#include "brave/components/brave_wallet/browser/brave_wallet_service.h"
+#include "content/public/browser/web_contents.h"
 
 namespace brave_wallet {
 
 void ShowPanel(content::WebContents* web_contents) {
-  NOTIMPLEMENTED();
+  BraveWalletService* brave_wallet_service =
+      brave_wallet::BraveWalletServiceFactory::GetServiceForContext(
```

Comment:
we shouldn't be using `GetServiceForContext` outside the component and factories, those methods were really meant to be private with friend classes

Reply:
should be using `BindForContext` everywhere else

Reply:
I'm going to correct it in a follow up https://github.com/brave/brave-browser/issues/22056